Kickstarter-iOS/Views/Controllers/LiveStreamChatViewController.swift

```diff
@@ -0,0 +1,268 @@
+import KsApi
+import Library
+import LiveStream
+import Prelude
+import ReactiveSwift
+import Result
+import UIKit
+
+fileprivate enum Section: Int {
+  case messages
+}
+
+internal protocol LiveStreamChatViewControllerDelegate: class {
+  func willPresentMoreMenuViewController(controller: LiveStreamChatViewController,
+                                         moreMenuViewController: LiveStreamContainerMoreMenuViewController)
+  func willDismissMoreMenuViewController(controller: LiveStreamChatViewController,
+                                         moreMenuViewController: LiveStreamContainerMoreMenuViewController)
+}
+
+internal final class LiveStreamChatViewController: UIViewController {
+
+  @IBOutlet private weak var tableView: UITableView!
+  @IBOutlet private weak var chatInputViewContainer: UIView!
+  @IBOutlet private weak var chatInputViewContainerBottomConstraint: NSLayoutConstraint!
+  @IBOutlet private weak var chatInputViewContainerHeightConstraint: NSLayoutConstraint!
+
+  fileprivate let dataSource = LiveStreamChatDataSource()
+  fileprivate weak var delegate: LiveStreamChatViewControllerDelegate?
+  fileprivate weak var liveStreamChatHandler: LiveStreamChatHandler?
+  fileprivate let shareViewModel: ShareViewModelType = ShareViewModel()
+  fileprivate let viewModel: LiveStreamChatViewModelType = LiveStreamChatViewModel()
+
+  public static func configuredWith(
+    delegate: LiveStreamChatViewControllerDelegate,
+    project: Project,
+    liveStreamEvent: LiveStreamEvent,
+    liveStreamChatHandler: LiveStreamChatHandler) ->
+    LiveStreamChatViewController {
+
+      let vc = Storyboard.LiveStream.instantiate(LiveStreamChatViewController.self)
+      vc.delegate = delegate
+      vc.liveStreamChatHandler = liveStreamChatHandler
+      vc.viewModel.inputs.configureWith(project: project, liveStreamEvent: liveStreamEvent, chatHidden: false)
+      vc.shareViewModel.inputs.configureWith(shareContext: .liveStream(project, liveStreamEvent))
+
+      return vc
+  }
+
+  internal override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.tableView.dataSource = self.dataSource
+
+    self.tableView.transform = CGAffineTransform(scaleX: 1, y: -1)
```

Comment:
I know this is a cheeky hack but it really simplifies an otherwise annoying problem with table views! By flipping the table view on its head and inversely flipping all of its cells we get the following for free:

- Automatically pins the table view to the bottom for new messages being inserted at row `0`.
- Animates the table view up with the keyboard appearance.
- Keeps the `contentOffset` on rotation.
- It's really simple.

The alternative (and my prior attempt) was a bunch of inputs and outputs to the view model handling all of these events and scrolling to the bottom on new messages, etc. but it would sometimes get the offset wrong after rotation or not animate smoothly with the keyboard appearance, etc.

I welcome other/better approaches!